From 4ddaf9f7a43ce9088018d9117376b56e10bbdd06 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Thu, 18 Sep 2014 14:14:07 -0400
Subject: [scsi] megaraid_sas: Load correct raid context timeout

Message-id: <1411049668-28841-8-git-send-email-thenzl@redhat.com>
Patchwork-id: 96151
O-Subject: [RHEL7.1 e-stor PATCH 07/28] megaraid_sas: Load correct raid context timeout
Bugzilla: 1088523
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

The following patch for megaraid_sas loads the correct raid context timeout
value for multpathing and clustering.

Signed-off-by: Adam Radford <aradford@gmail.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 26a077e7a6b471926da3a70b7fb212758463f589)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/megaraid/megaraid_sas_fp.c b/drivers/scsi/megaraid/megaraid_sas_fp.c
index 2f3bd55..081bfff 100644
--- a/drivers/scsi/megaraid/megaraid_sas_fp.c
+++ b/drivers/scsi/megaraid/megaraid_sas_fp.c
@@ -975,7 +975,10 @@ MR_BuildRaidContext(struct megasas_instance *instance,
    regSize += stripSize;
  }
 
- pRAID_Context->timeoutValue     = cpu_to_le16(map->raidMap.fpPdIoTimeoutSec);
+ pRAID_Context->timeoutValue =
+  cpu_to_le16(raid->fpIoTimeoutForLd ?
+       raid->fpIoTimeoutForLd :
+       map->raidMap.fpPdIoTimeoutSec);
  if ((instance->pdev->device == PCI_DEVICE_ID_LSI_INVADER) ||
   (instance->pdev->device == PCI_DEVICE_ID_LSI_FURY))
   pRAID_Context->regLockFlags = (isRead) ?
-- 
1.7.1